pkg/experimental/tests/pkg_rpm_basic_test.py

```diff
@@ -0,0 +1,131 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import subprocess
+import csv
+import io
+from rules_python.python.runfiles import runfiles
+
+# This provides some tests for built RPMs, mostly by taking the built RPM and
+# running rpm queries on it.
+#
+# Useful reading:
+#
+# - RPM queryformat documentation (shortish):
+#   https://rpm.org/user_doc/query_format.html
+#
+# - In-depth RPM query documentation:
+#   http://ftp.rpm.org/max-rpm/s1-rpm-query-parts.html
+#
+# - Specifically, about the --qf/--queryformat syntax:
+#   http://ftp.rpm.org/max-rpm/s1-rpm-query-parts.html#S3-RPM-QUERY-QUERYFORMAT-OPTION
+#
+# - --queryformat tags list: http://ftp.rpm.org/max-rpm/ch-queryformat-tags.html
+#
+class PkgRpmBasicTest(unittest.TestCase):
+    def setUp(self):
+        self.runfiles = runfiles.Create()
+        self.test_rpm_path = self.runfiles.Rlocation(
+            "rules_pkg/experimental/tests/test_rpm.rpm")
+        self.maxDiff = None
+
+    def test_scriptlet_content(self):
+        expected = b"""\
+preinstall scriptlet (using /bin/sh):
+echo pre
+postinstall scriptlet (using /bin/sh):
+echo post
+preuninstall scriptlet (using /bin/sh):
+echo preun
+postuninstall scriptlet (using /bin/sh):
+echo postun
+"""
+
+        output = subprocess.check_output(
+            ["rpm", "-qp", "--scripts", self.test_rpm_path])
+
+        self.assertEqual(output, expected)
+
+    def test_basic_headers(self):
+        fields = {
+            "NAME": b"test_rpm",
+            "VERSION": b"1.1.1",
+            "RELEASE": b"2222",
+            "ARCH": b"noarch",
+            "GROUP": b"Unspecified",
+            "SUMMARY": b"pkg_rpm test rpm summary",
+        }
+        for fieldname, expected in fields.items():
+            output = subprocess.check_output([
+                "rpm", "-qp", "--queryformat", "%{" + fieldname + "}",
+                self.test_rpm_path
+            ])
+
+            self.assertEqual(
+                output, expected,
+                "RPM Tag {} does not match expected value".format(fieldname))
+
+    def test_contents(self):
+        self.manifest_file = self.runfiles.Rlocation(
+            "rules_pkg/experimental/tests/manifest.txt")
+        manifest_specs = {}
+        with open(self.manifest_file, "r", newline='', encoding="utf-8") as fh:
+            manifest_reader = csv.DictReader(fh)
+            manifest_specs = {r['path']: r for r in manifest_reader}
+
+        # The second one would be %{FILESIZES}, but the size in the RPM changes
+        # a little, probably due to rpm's internal "magic".
+        # TODO(nacl): investigate this
+        #
+        # Documentation on this:
+        # - Query format syntax:
+        rpm_queryformat = \
+            "[%{FILENAMES}" + \
```

Comment:
The + is not needed along with the \\
This would be more readable as
  rpm_queryformat = (
     "[%{FILENAMES}" 
     ",%{FILEDIGESTS}" 
     ...
    "\\n]")

Reply:
I totally didn't know you could do this.  Will fix, thanks!

Reply:
Fixed in 2a74ca2